.github/workflows/publish-docker-image.yml

```diff
@@ -0,0 +1,78 @@
+name: Build and Publish Docker Image
+
+on:
+  workflow_call:
+    inputs:
+      flavor:
+        description: "Defines global behaviors for tags (e.g. 'latest=auto')"
+        required: false
+        type: string
+
+      images:
+        description: "Docker image names (e.g. 'ghcr.io/FuelLabs/repository-name')"
+        required: true
+        type: string
+
+      labels:
+        description: "Docker image labels (e.g. 'com.example.label=value')"
+        required: false
+        type: string
+
+      tags:
+        description: "Docker image tags (e.g. 'latest,1.0.0')"
+        required: true
+        type: string
+
+    secrets:
+      GITHUB_CONTAINER_USERNAME:
+        description: "Username for GitHub Container Registry"
+        required: true
+
+      GITHUB_CONTAINER_PASSWORD:
+        description: "Password for GitHub Container Registry"
+        required: true
+
+env:
+  GITHUB_CONTAINNER_REGISTRY_URL: ghcr.io
+
+jobs:
+  build-and-publish-docker-image:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Docker Meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          flavor: ${{ inputs.flavor }}
+          images: ${{ inputs.images }}
+          labels: ${{ inputs.labels }}
+          tags: ${{ inputs.tags }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.GITHUB_CONTAINNER_REGISTRY_URL }}
+          username: ${{ secrets.GITHUB_CONTAINER_USERNAME }}
+          password: ${{ secrets.GITHUB_CONTAINER_PASSWORD }}
+
+      - name: Build and publish image to Github Container Registry
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: deployment/Dockerfile
```

Comment:
We should make this a configurable input, as some projects could have multiple dockerfiles to publish etc

Reply:
Just added an input with a default value of `deployment/Dockerfile`